mondey_backend/src/mondey_backend/routers/users.py

```diff
@@ -78,12 +81,32 @@ def update_child(
 
     @router.delete("/children/{child_id}")
     def delete_child(
-        session: SessionDep, current_active_user: CurrentActiveUserDep, child_id: int
+        session: SessionDep,
+        current_active_user: CurrentActiveUserDep,
+        child_id: int,
+        dry_run: bool = Query(
+            True,
+            description="When true, shows what would be deleted without actually deleting",
+        ),
     ):
         child = get_db_child(session, current_active_user, child_id)
+        # the above guards it to only owners of the child.
+        if dry_run:
+            affectedAnswers = 0
+            for answering_session in get_childs_answering_sessions(session, child_id):
+                affectedAnswers += len(answering_session.answers)
+            return {
+                "ok": True,
+                "would_delete": {"affectedAnswers": affectedAnswers},
+            }
+
+        delete_statement = delete(MilestoneAnswerSession).where(
+            col(MilestoneAnswerSession.child_id) == child_id
+        )
+        session.execute(delete_statement)
```

Comment:
we should also remove the child image if it exists here:
`child_image_path(child_id).unlink(missing_ok=True)`